homeassistant/components/tplink/button.py

```diff
@@ -0,0 +1,69 @@
+"""Support for TPLink button entities."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Final
+
+from kasa import Feature
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import TPLinkConfigEntry
+from .entity import CoordinatedTPLinkFeatureEntity, TPLinkFeatureEntityDescription
+
+
+@dataclass(frozen=True, kw_only=True)
+class TPLinkButtonEntityDescription(
+    ButtonEntityDescription, TPLinkFeatureEntityDescription
+):
+    """Base class for a TPLink feature based button entity description."""
+
+
+BUTTON_DESCRIPTIONS: Final = [
+    TPLinkButtonEntityDescription(
+        key="test_alarm",
+    ),
+    TPLinkButtonEntityDescription(
+        key="stop_alarm",
+    ),
+]
+
+BUTTON_DESCRIPTIONS_MAP = {desc.key: desc for desc in BUTTON_DESCRIPTIONS}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: TPLinkConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up buttons."""
+    data = config_entry.runtime_data
+    parent_coordinator = data.parent_coordinator
+    children_coordinators = data.children_coordinators
+    device = parent_coordinator.device
+
+    entities = CoordinatedTPLinkFeatureEntity.entities_for_device_and_its_children(
+        device=device,
+        coordinator=parent_coordinator,
+        feature_type=Feature.Type.Action,
+        entity_class=Button,
+        descriptions=BUTTON_DESCRIPTIONS_MAP,
+        child_coordinators=children_coordinators,
+    )
+    async_add_entities(entities)
+
+
+class Button(CoordinatedTPLinkFeatureEntity, ButtonEntity):
```

Comment:
We should call these `TPLinkButtonEntity` so when we are debugging we won't have to guess which integration `Button` objects belong to when looking at memory/gc/objects.

Reply:
Done in https://github.com/home-assistant/core/pull/120404